packages/asset-service/src/service/AssetService.ts

```diff
@@ -12,15 +14,35 @@ export const flattenAssetData = (assetData: BaseAsset[]): Asset[] => {
     flatAssetData.push(newAsset)
     if (baseAsset.tokens) {
       for (const tokenAsset of baseAsset.tokens) {
-        flatAssetData.push({
+        const flatAssetDatum = {
           ...tokenAsset,
           chain: baseAsset.chain,
           network: baseAsset.network,
           slip44: baseAsset.slip44,
           explorer: baseAsset.explorer,
           explorerAddressLink: baseAsset.explorerAddressLink,
           explorerTxLink: baseAsset.explorerTxLink
-        })
+        }
+        if (!flatAssetDatum.icon) {
+          const { tokenId } = caip19.fromCAIP19(flatAssetDatum.caip19)
+          const options: IdenticonOptions = {
+            identiconImage: {
+              size: 128,
+              background: [45, 55, 72, 255]
+            },
+            identiconText: {
+              symbolScale: 7,
+              enableShadow: true
+            }
+          }
+          if (tokenId)
+            flatAssetDatum.icon = getRenderedIdenticonBase64(
```

Comment:
This is being called during app launch. We measured the time to generate an icon at around 30ms. So, if there are a substantial number of assets without icons, this loop could cause a significant delay in app startup time.

I'm discussing with some other core devs about whether we should put this in the generate script so it happens at build time, or have the `AssetIcon` component leverage the function to generate the icon only at display time.

I'm open to thoughts on these or other approaches to reduce the time to first load of the app.

Reply:
@Soptq Major Hayes makes a good point. I pegged the icon generation at 30ms per icon. If we have a very large set of assets without an icon, say 1000, then that would add 30sec to the initial app startup.

Can you move the logic implementation from here to the asset generation files in asset-service/generateAssetData/ ethTokens and cosmos?

Reply:
adding another +1 to this, especially if we can just base64 them into the static data. 

the other option is to expose this `getRenderedIdenticonBase64` function that web can call at runtime to generate them as components mount, rather than at app load - i don't expect generating these is going to be any slower than a network request to load an icon, and it keeps the initial payload smaller.